File: Calculo-Computacional/Tarea-2/Codigos/pregunta2.py
import metodos
from math import cos, sin, pi, exp

def f(x):
	return 0.25 * (x**2) - x * sin(x) + 0.5 * -cos(2*x) + 0.5

def d(x):
	return  x * (0.5 - cos(x)) - sin(x) + sin(2*x)

if __name__ == '__main__':
	estimados = [pi/2, 5*pi, 10*pi]
	for p0 in estimados:
		print '-------------------------------------'
		value = metodos.newton(p0, f, d)
		print p0, ", Final Value: %.7f" % (value)

File: Calculo-Computacional/Tarea-2/Codigos/pregunta2.cpp
#include "metodos.hpp"
#include <iostream>
#include <iomanip>
#include <vector>
#include <cmath>
using namespace std;

double f(double x){
	return 0.25 * pow(x, 2) - x * sin(x) - 0.5 * cos(2*x) + 0.5;
}

double d(double x){
	return 0.5 * x - sin(x) - x * cos(x) + sin(2*x);
}

int main(){
	double arr[3]= {M_PI/2, 5*M_PI, 10*M_PI};
	vector<double> estimados(&arr[0], &arr[0]+3);
	double solucion;
	
	cout << std::setprecision(8);
	for(int i=0; i < (int)estimados.size(); i++){
		cout << "Probando con " << estimados[i] << " -- ";
		solucion = newton(estimados[i], f, d);
		cout << "Final value: " << solucion << endl;
	}
	
	return 0;
}

